src/modules/0.0.22/aws_memory_db/sql/after_install.sql

```diff
@@ -27,3 +27,43 @@ UPDATE
   ON subnet FOR EACH ROW
 EXECUTE
   FUNCTION check_subnets_by_subnet_group ();
+
+-- Since Postgres does not allow arrays of foreign keys, but we prefer the syntactic simplicity they
+-- provide. This check function implements half of a foreign key's behavior by making sure on insert
+-- or update of a memory db subnet group that all referenced subnets actually exist.
+CREATE
+OR REPLACE FUNCTION check_subnet_group_subnets () RETURNS TRIGGER LANGUAGE plpgsql SECURITY DEFINER AS $check_subnet_group_subnets$
+declare
+  _subnets_count integer;
+  _vpc_count integer;
+begin
+  select COUNT(*) into _subnets_count
+  from subnet
+  where subnet_id = any(NEW.subnets) AND region = NEW.region;
+
+  select COUNT(distinct vpc_id) into _vpc_count
+  from subnet
+  where subnet_id = any(NEW.subnets) and region = NEW.region;
+
+  IF ((_subnets_count = 0 OR _subnets_count = array_length(NEW.subnets, 1)) AND _vpc_count <= 1) THEN
+    RETURN NEW;
+  ELSE
+    RAISE EXCEPTION 'Subnets must be from the same vpc and region.';
+    RETURN OLD;
+  END IF;
+end;
+$check_subnet_group_subnets$;
+
+CREATE TRIGGER
+  check_subnet_group_subnets BEFORE INSERT
+  OR
+UPDATE
+  ON subnet_group FOR EACH ROW
+EXECUTE
+  FUNCTION check_subnet_group_subnets ();
+
+-- We have a weird case were we need to add 2 FKs to a join table column, and typeorm is not that flexible to define that
+ALTER TABLE
+  "memory_db_cluster_security_groups"
+ADD
+  CONSTRAINT "FK_memory_db_cluster_security_groups_region" FOREIGN KEY ("security_group_id", "region") REFERENCES "security_group" ("id", "region") ON DELETE CASCADE ON UPDATE CASCADE;
```

Comment:
This is actually not true. We figured out how to do it. We need to add the `@unique` on `id` and `region` in the `SecurityGroup` entity and then we can do the `@JoinColumn([...])` syntax.

Reply:
I tried that, but it is a join table with join columns and Typeform does not like them having the same name. It renames it like `region_1` and `region_2`. I also tried adding two different columns each one referencing its region but was having trouble with "duplicated" inserts in the join table and couldn't figure out why 🤔